languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml

```diff
@@ -11906,9 +11881,193 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
                 <example>Torna-se necessário obedecer a normas de conduta.</example>
                 <example>A Literatura obedece a leis inflexíveis: a da herança, a do meio, a do momento.</example>
             </rule>
-        </rulegroup>
+            <!-- Moved from the PHRASAL_VERBS_ERRORS rulegroup above and split to handle contractions. -->
+            <rule>
+                <antipattern>
+                    <token min='0' regexp='yes'>me|te|se|vos</token>
+                    <token inflected='yes'>habituar</token>
+                    <token min='0' regexp='yes' spacebefore='no'>&hifen;</token>
+                    <token min='0' regexp='yes'>me|te|se|nos|vos</token>
+                    <token>com</token>
+                    <token regexp="yes">a(quel[ea])?s?|os?</token>
+                </antipattern>
+                <antipattern>
+                    <token min='0' regexp='yes'>me|te|se|vos</token>
+                    <token inflected='yes'>habituar</token>
+                    <token min='0' regexp='yes' spacebefore='no'>&hifen;</token>
+                    <token min='0' regexp='yes'>me|te|se|nos|vos</token>
+                    <token regexp="yes">co(m|n[st])igo|conv?osco</token>
+                </antipattern>
+                <pattern>
+                    <token min='0' regexp='yes'>me|te|se|vos</token>
+                    <token inflected='yes'>habituar</token>
+                    <token min='0' regexp='yes' spacebefore='no'>&hifen;</token>
+                    <token min='0' regexp='yes'>me|te|se|nos|vos</token>
+                    <marker>
+                        <token>com</token>
+                    </marker>
+                </pattern>
+                <message>Neste contexto, o verbo &quot;habituar&quot; rege-se com a preposição &quot;a&quot;.</message>
+                <suggestion>a</suggestion>
+                <example correction='a'>Ainda não me habituei <marker>com</marker> ele.</example>
+                <example correction='a'>Tenho dificuldades em habituar-me <marker>com</marker> novas rotinas.</example>
+                <example>Consegues habituar-te com calma ao clima.</example>
```

Comment:
There is an antipattern/exception missing for `<example>Consegues habituar-te com calma ao clima.</example>`.

Reply:
<img width="206" alt="Screenshot 2023-05-22 at 10 25 53 AM" src="https://github.com/languagetool-org/languagetool/assets/50704700/e46769fd-b30e-44f0-8190-7b3938794195">

Reply:
I tested this locally, and the error is still triggered for this sentence:

```
org.languagetool.rules.patterns.PatternRuleTest$PatternRuleTestFailure: Test failure for rule PHRASAL_VERB_A2[1] in file /org/languagetool/rules/pt/style.xml (line 21639): Did not expect error in:
  Consegues habituar-te com calma ao clima.
  Analyzed token readings: [/SENT_START*] Consegues[conseguir/VMIP2S0*]  [ /null*] habituar[habituar/VMN0000,habituar/VMN01S0,habituar/VMN03S0,habituar/VMSF1S0,habituar/VMSF3S0] -[-/_PUNCT*] te[te/PP2CS000*]  [ /null*] com[com calma/RG]  [ /null*] calma[com calma/RG]  [ /null*] ao[a:o/SPS00:DA0MS0]  [ /null*] clima[clima/NCMS000] .[./SENT_END*,./_PUNCT_PERIOD*,./_PUNCT*]
```

So it needs `<token negate="yes">calma</token>` after the marker. The rule editor isn't always accurate. In this case, it could have something to do with the multi-word tagging of `com calma` (tagged as `RG`, not `SPS00` + `NCFS000`).

You can search for `PortuguesePatternRuleTest` on Intellij to run the test locally.

Reply:
Right, I *did* run the local tests earlier, though, and everything passed. I've just tried changing `calma` to something else *and only then* does it actually fail. Likewise, the same tests were run on (Circle)[https://app.circleci.com/pipelines/github/languagetool-org/languagetool/22048/workflows/efc218cc-351e-404d-8fcf-175792e2c025/jobs/22182], and they're also fine 🤔 Are you running the tests on my branch?

It will probably not *hurt* to negate `calma`, sure, it just doesn't seem like things work very consistently for this one construction and I couldn't find anything specific to it... I mean, there are tonnes of adverbial phrases beginning with `com` and apparently they're tagged differently.

Reply:
It's done, along with a few other common nouns to be found after `com` in adverbial phrases.

Reply:
Ah, found the source. I only ran the test on your changes, not the whole rule. 
`com calma` (`RG` + `RG`) apparently was the only exception that worked because of this antipattern, which is valid for the entire rule group:

 ```
<antipattern>
         <token postag_regexp='yes' postag='R.+'>
          <exception negate_pos='yes' postag_regexp='yes' postag='R.+'/></token>
          <token postag_regexp='yes' postag='R.+'>
          <exception negate_pos='yes' postag_regexp='yes' postag='R.+'/></token>
</antipattern>
```

We are good to go. We should get some diff results by Wednesday, so perfect timing 😄